### DISCUSSION TITLE
[Proposal] Add edge detection for document capture using the camera view.

### DISCUSSION THREAD

**Discussion by mancda** [CommunityToolkit/Maui#2116] [New Feature Discussions]
### Feature name

CameraView Edge detection

### Link to discussion

No link

### Progress tracker

- [X] Android Implementation
- [X] iOS Implementation
- [ ] MacCatalyst Implementation
- [ ] Windows Implementation
- [ ] Tizen Implementation
- [ ] Unit Tests
- [ ] Samples
- [ ] Documentation

### Summary

It would be wonderful to be able to use the camera view to scan documents using the phone.

### Motivation

Would be helpful for applications that need to capture documents in the field

### Detailed Design

For an example of edge detection see Adobe scanning

### Usage Syntax

```markdown
Not applicable
```

**Comment by MartinRobins** [comment]
Could this potentially be implemented as a "scenario" as described in [CameraView frame processing support #1687 here](https://github.com/CommunityToolkit/Maui/discussions/1687)?
I too am looking at ways to achieve this. I have an android implementation already based upon the older camera APIs, but would like to look at updating that to the new APIs before the older ones are removed altogether.

**Reply by bijington** [reply]
I would like to be able to support scenarios like yours. I believe Android is setup well to support it. I am not convinced iOS is though. I don't want to rush this implementation though so adding scenarios like this is great 👍
